pkg/apis/serving/k8s_validation.go

```diff
@@ -267,7 +277,60 @@ func ValidatePodSpec(ps corev1.PodSpec) *apis.FieldError {
 	return errs
 }
 
+//ValidateMultiContainerPorts validates port when specified multiple containers
+func ValidateMultiContainerPorts(containers []corev1.Container) *apis.FieldError {
+	var (
+		count int
+		errs  *apis.FieldError
+	)
+	for i := range containers {
+		for range containers[i].Ports {
+			count++
+		}
+	}
+	if count == 0 {
+		errs = errs.Also(apis.ErrMissingField(apis.CurrentField))
+	}
+	return errs.Also(portValidation(count))
+}
+
+//ValidateSidecarContainer validate fields for non serving containers
+func ValidateSidecarContainer(container *corev1.Container, volumes sets.String) *apis.FieldError {
```

Comment:
This should probably take just `corev1.Container` like the rest of this file so you don't need to pass a reference or not a reference in `ValidatePodSpec`.